src/clj/rems/db/applications.clj

```diff
@@ -282,11 +283,15 @@
   :start (scheduler/start! reload-cache! (Duration/standardHours 1))
   :stop (scheduler/stop! all-applications-cache-reloader))
 
+(defn remove-application-data!
```

Comment:
One option for the names would be `delete-application!` and `delete-application-and-reload-cache!`. Just something completely explicit. Now they are sort of close and I wouldn't be sure what they actually do without reading the source. We could also consider adding docstrings.